plugins/obs-ffmpeg/obs-ffmpeg-source.c

```diff
@@ -315,6 +301,44 @@ static void ffmpeg_source_start(struct ffmpeg_source *s)
 	}
 }
 
+static void *ffpmeg_source_reconnect(void *data)
+{
+	os_sleep_ms(10000);
+	struct ffmpeg_source *s = data;
+	FF_BLOG(LOG_INFO, "MP: start reconnect thread");
+	if (s->media_valid) {
+		return NULL;
+	}
+	bool active = obs_source_active(s->source);
+	if (!s->close_when_inactive || active)
+		ffmpeg_source_open(s);
+
+	if (!s->restart_on_activate || active)
+		ffmpeg_source_start(s);
+	return NULL;
+}
+
+static void ffmpeg_source_tick(void *data, float seconds)
+{
+	UNUSED_PARAMETER(seconds);
+
+	struct ffmpeg_source *s = data;
+	if (s->destroy_media) {
+		if (s->media_valid) {
+			mp_media_free(&s->media);
+			s->media_valid = false;
+		}
+		s->destroy_media = false;
+		FF_BLOG(LOG_ERROR, "MP: thread has been destory");
+		pthread_t reconnect_thread;
+		if (pthread_create(&reconnect_thread, NULL,
```

Comment:
This creates a dangling thread. If the source is destroyed, the thread is not correctly signaled to exit and waited on, which will result in a crash.